pages/admin/index.tsx

```diff
@@ -104,44 +117,82 @@ const COLUMNS = [
   },
 ];
 
-// Placeholder data
-const DATA = [
-  {
-    name: {
-      name: 'Steve Rogers',
-      id: '36565',
-    },
-    dateReceived: 'Dec 21 2021, 8:30 pm',
-    permitType: 'Permanent',
-    requestType: 'Replacement',
-    status: 'PENDING',
-  },
-  {
-    name: {
-      name: 'Steve Rogers',
-      id: '36565',
-    },
-    dateReceived: 'Dec 21 2021, 8:30 pm',
-    permitType: 'Permanent',
-    requestType: 'Replacement',
-    status: 'INPROGRESS',
-  },
-  {
-    name: {
-      name: 'Steve Rogers',
-      id: '36565',
-    },
-    dateReceived: 'Dec 21 2021, 8:30 pm',
-    permitType: 'Permanent',
-    requestType: 'Replacement',
-    status: 'COMPLETED',
-  },
-];
+// Application data for table
+type ApplicationData = {
+  name: {
+    name: string;
+    rcdUserId: number | undefined;
+  };
+  dateReceived: Date;
+  permitType: string;
+  requestType: string;
+  status: ApplicationStatus | undefined;
+};
+
+// Max number of entries in a page
+const PAGE_SIZE = 20;
 
 // Internal home page - view APP requests
 export default function Requests() {
-  const [permitTypeFilter, setPermitTypeFilter] = useControllableState({ defaultValue: 'All' });
-  const [requestTypeFilter, setRequestTypeFilter] = useControllableState({ defaultValue: 'All' });
+  //Filters
+  const [statusFilter, setStatusFilter] = useState<ApplicationStatus>();
+  const [permitTypeFilter, setPermitTypeFilter] = useState<PermitType>();
+  const [requestTypeFilter, setRequestTypeFilter] = useState<string>();
+  const [searchFilter, setSearchFilter] = useState<string>('');
+
+  // Sorting
+  const [sortOrder, setSortOrder] = useState<SortOptions>([['dateReceived', SortOrder.DESC]]);
+
+  // Debounce search filter so that it only gives us latest value if searchFilter has not been updated within last 500ms.
+  // This will avoid firing a query for each key the user presses
+  const debouncedSearchFilter = useDebounce<string>(searchFilter, 500);
+
+  // Data & pagination
+  const [requestsData, setRequestsData] = useState<ApplicationData[]>();
+  const [pageNumber, setPageNumber] = useState(0);
+  const [recordsCount, setRecordsCount] = useState(0);
+
+  // Make query to applications resolver
+  useQuery<GetApplicationsResponse, GetApplicationsRequest>(GET_APPLICATIONS_QUERY, {
```

Comment:
Following the two comments above, we can grab the query result like this:
```suggestion
  const { data } = useQuery<GetApplicationsResponse, GetApplicationsRequest>(GET_APPLICATIONS_QUERY, {
```
and we can destructure the `results` and `totalCount` from `data`

Reply:
We actually can't do this because the data value is never used. The `data` that's used inside onCompleted doesn't correspond to the same data variable so it produces errors